types/d3-scale/d3-scale-tests.ts

```diff
@@ -78,9 +78,11 @@ tickFormatNumberFn = d3Scale.tickFormat(0, 10, 5, '+%');
 
 // scaleLinear() -----------------------------------------------------------------
 
-let linearScaleNumber: d3Scale.ScaleLinear<number, number>;
-let linearScaleString: d3Scale.ScaleLinear<string, string>;
-let linearScaleNumString: d3Scale.ScaleLinear<number, string>;
+let linearScaleNumber: d3Scale.ScaleLinear<number, number, never>;
```

Comment:
I wonder if this is going to break things for existing users, too, since `ScaleLinear<number, number>` is now an error.

Since you can't specify some generic parameters but not others, a workaround would be to introduce a new type and replace the existing one with something like:

```ts
type ScaleLinear<Domain, Range> = ScaleLinearUnknown<Domain, Range, never>;
```

Reply:
How often do users of the types reference the interfaces directly? My understanding is that most people use the top-level methods (which have friendly generic defaults) and do most things inline., so there's not usually a need to write out the interface type. Also, as a general pattern for `d3-*` types we don't specify defaults for the interface generics and instead provide generic defaults for the functions to make them easy-to-use.

Reply:
Anecdotally, all of the TS files in my project that use d3-scale reference `ScaleLinear` and will break with this.

If you define a class and store the scale in a member variable, or if you're working in React and want to store it in state, you need to write out the type. See https://github.com/microsoft/TypeScript/issues/33272 for previous problems that changing generic behavior caused with d3 (those were TS changes, not @types/d3 changes, but it shows that these types are very much in use).

Reply:
I went ahead and made the change.